commerce_wishlist.module

```diff
@@ -192,3 +194,137 @@ function commerce_wishlist_order_type_form_submit($form, FormStateInterface $for
+      '#default_value' => $plugin->getThirdPartySetting('commerce_wishlist', 'show_wishlist'),
+    );
+    $element['ajax_wishlist'] = array(
+      '#type' => 'checkbox',
+      '#title' => t('Display "Add to wishlist" as an ajax link'),
+      '#default_value' => $plugin->getThirdPartySetting('commerce_wishlist', 'ajax_wishlist'),
+    );
+    $element['weight_wishlist'] = array(
+      '#type' => 'number',
+      '#title' => t('Change the weight of the wishlist button.'),
+      '#default_value' => $plugin->getThirdPartySetting('commerce_wishlist', 'weight_wishlist'),
+    );
+    $element['label_wishlist'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Override the wishlist button label'),
+      '#default_value' => $plugin->getThirdPartySetting('commerce_wishlist', 'label_wishlist'),
+    );
+  }
+  return $element;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function commerce_wishlist_form_commerce_line_item_product_variation_add_to_cart_form_alter(&$form, \Drupal\Core\Form\FormStateInterface $form_state, $form_id) {
+
+  // The product entity is thankfully available in the form storage.
+  $form_data = $form_state->getStorage();
+  /** @var \Drupal\commerce_product\Entity\ProductInterface $product */
+  $product = $form_data['product'];
+
+  // Grab the view mode third party settings.
+  $display = EntityViewDisplay::collectRenderDisplay($product, $form_data['view_mode']);
+  $settings = $display->get("content")['variations']['third_party_settings']['commerce_wishlist'];
+
+  // Add the button
+  if ($settings['show_wishlist'] == 1) {
+    $label = ($settings['label_wishlist'] != "")? $settings['label_wishlist'] : t('Add to wishlist');
+    $weight = ($settings['weight_wishlist'] != "")?$settings['weight_wishlist']:99;
+    if ($settings['ajax_wishlist'] != 1) {
+      $form['actions']['wishlist'] = array(
+        '#type' => 'submit',
+        '#value' => $label,
+        '#weight' => $weight,
+        '#submit' => array('commerce_wishlist_add_to_wishlist_form_submit'),
+      );
+    } else {
+      $form['actions']['wishlist'] = array(
+        '#markup' => '<a href="" class="commerce-wishlist-ajax">' . $label . ' - Not Implemented Yet</a>',
+        '#weight' => $weight,
+      );
+    }
+  }
+}
+
+/**
+ * Form submit handler for add-to-wishlist actions.
+ *
+ * @param array $form
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ */
+function commerce_wishlist_add_to_wishlist_form_submit($form, FormStateInterface $form_state) {
+  \Drupal::logger('commerce_wishlist')->notice('for has been submitted.') ;
+  /** @var \Drupal\commerce_order\Entity\LineItemInterface $line_item */
+  $line_item = $form_state->getFormObject()->getEntity();
+  /** @var \Drupal\commerce\PurchasableEntityInterface $purchased_entity */
+//  $purchased_entity = $line_item->get;
+  ob_start();
+  var_dump($line_item);
+
+  \Drupal::logger('commerce_wishlist')->notice("hello? ".ob_get_clean());
+  return;
```

Comment:
By the way, if you're reading this then you might go "what the heck?" ... I agree with you. Definitely a WIP ... but I'm seriously lost at this point and about to abandon hope of a simple solution here.